### PR TITLE
Rmadeye

### PR DESCRIPTION
**Pull request by Rmadeye**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,38 @@
...
+
+1. Clone repository
+```
+git clone https://github.com/labstructbioinf/dc2_oligo
+```
+
+2. Install all dependencies
+
+```
+pip install scikit-learn pandas joblib ...
```
Comment (jludwiczak) [inline README.md]: I'm more in favor of specifying the conda yaml file with all dependencies and possibly their version.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,38 @@
...
+```
+
+3. Check if everything works using pytest
+
+```bash
+cd dc2_oligo
+pytest
```
Comment (jludwiczak) [inline README.md:22]: `pytest` should be run from repo root I think, it'll discover tests in `test` directory and similar.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,38 @@
...
+python src/predict.py --colabfold_output_dir DIR --use_pairwise BOOL
+
+ ```
+ | Argument        | Description |
+|:-------------:|-------------|
+| **`--colabfold_output_dir`** | Colabfold output directory with saved embeddings via --save-representations option |
+| **`--use_pairwise`** | Flag for using sorting pair representations
+
+```bash
+python src/predict.py --colabfold_output_dir tests/data/7 --use_pairwise True
```
Comment (jludwiczak) [inline README.md]: Move it from src to repo root, it's more straightforward and common, alternatively keep it in source but make an entrypoint to run it as standalone script - e.g. https://github.com/labstructbioinf/localpdb/blob/master/pyproject.toml#L20-L21

File: src/data/set4_homooligomers.csv
Comment (jludwiczak) [inline src/data/set4_homooligomers.csv:1]: Why `data` is kept in `src` ?

File: src/predict.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import numpy as np
+import argparse
+import glob
+import joblib
+import sys
+import os
+root_directory = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+
+sys.path.append(root_directory)
```
Comment (jludwiczak) [inline src/predict.py]: To dziala na mnie jak plachta na byka :D

File: src/results/model.p
Comment (jludwiczak) [inline src/results/model.p:1]: Move away from src

File: tests/test_dc2.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+from src.predict import predict_oligo_state
+
+# @pytest.mark.parametrize("use_pairwise", [True, False])
+def test_predict_oligo_state():
+    result_1 = predict_oligo_state("tests/data/7/", use_pairwise=True)
+    result_2= predict_oligo_state("tests/data/10/", use_pairwise=True)
+    # Add your assertions here to verify that the result is as expected
+    assert result_1 == 0
+    assert result_2 == 1
```
Comment (jludwiczak) [inline tests/test_dc2.py]: This only checks whether predicted class is correct. It'd be good to check whether actual ensemble mean and stdev match too.

File: src/predict.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    if use_pairwise:
+        mat = np.hstack((mat, np.load(pair_repr_fns[0]).mean(axis=0).mean(axis=0)))
+    
+    return mat
+
+def predict_oligo_state(colabfold_output_dir:  str, use_pairwise: bool):
+    model = joblib.load(f'{root_directory}/src/results/model.p')#['clf_0_4_3']  # !!!!!!!!!!!!!!!!!!!
+    X = np.asarray([get_af2_emb_upd(colabfold_output_dir, use_pairwise=use_pairwise)]).reshape(1, -1)
+    print(model)
+    output = model.predict(X)[0]
```
Comment (jludwiczak) [inline src/predict.py]: Here you should compute ensemble mean and standard deviation and report them along with the classification result.

File: src/train.py [hunk 7]
```diff
@@ -46,36 +48,34 @@ def train(args):
...
         for i in range(0, 5): # 5 since we have 5 AF2 models
 
-            X = np.asarray([get_af2_emb(id_, model_id=i, use_pairwise=args.use_pairwise) for id_ in df.index])
+            X = np.asarray([get_af2_emb(id_, model_id=i, use_pairwise=use_pairwise) for id_ in df.index])
             y = df['y'].values
 
-            cv = KFold(n_splits=5, shuffle=True)
-
-            for k, (tr_idx, te_idx) in enumerate(cv.split(X, y)):
+            X_tr, X_te, y_tr, y_te = train_test_split(X, y, test_size=0.2)
```
Comment (jludwiczak) [inline src/train.py]: We want to fit on whole dataset for deployment model, split is not needed here.

**Review by jludwiczak — CHANGES_REQUESTED** [review]
Looks good! Few general comments:
- Get rid of CV and train_test_split (fit whole dataset),
- Cleanup src dir, it's structure and imports,
- Predict with whole ensemble and report mean + stdev + argmax (already there),
- Reproducible env,
- Rest of minor things in comments.

File: README.md (hunk 2 above)
Comment (Rmadeye) [reply README.md:22]: It is in repo root, just suggested to first change dir into root directory after  cloning the repo

File: README.md (hunk 3 above)
Comment (Rmadeye) [reply README.md]: Added
